repose-aggregator/components/filters/keystone-v2-basic-auth-filter/src/main/scala/org/openrepose/filters/keystonev2basicauth/KeystoneV2BasicAuthFilter.scala

```diff
@@ -230,16 +229,14 @@ class KeystoneV2BasicAuthFilter @Inject()(configurationService: ConfigurationSer
             //Figure out what our Retry Headers are, if identity returned us a rate limited response.
             val retryHeaders: Option[String] = if (statusCode == HttpServletResponse.SC_REQUEST_ENTITY_TOO_LARGE | statusCode == SC_TOO_MANY_REQUESTS) {
               // (413 | 429)
-              val identityRetryHeader = tokenResponse.getHeaders.filter { header => header.getName.equals(HttpHeaders.RETRY_AFTER) }
-              if (identityRetryHeader.isEmpty) {
-                logger.info(s"Missing ${HttpHeaders.RETRY_AFTER} header on Auth Response status code: $statusCode")
-                val retryCalendar = new GregorianCalendar()
-                retryCalendar.add(Calendar.SECOND, 5)
-                val retryString = new HttpDate(retryCalendar.getTime).toRFC1123
-                Some(retryString)
-              } else {
-                Some(identityRetryHeader.head.getValue)
-              }
+              Some(
```

Comment:
I don't really like unpacking the `headOption` just to re pack it. Instead of `.getOrElse` just use `.orElse` it does what you want without it reading super weird.

Reply:
Makes sense.